app/models/blueprint.rb

```diff
@@ -1,4 +1,6 @@
 class Blueprint < ApplicationRecord
+  has_many :versions, :class_name => 'Blueprint',
+           :foreign_key => :original_blueprint_id, :primary_key => :original_blueprint_id
```

Comment:
Perhaps I'm reading it wrong, but the way it's implemented here there is basically 1 original, and every other version points to it, so it's not really a lineage, but a fanout.  @jntullo Am I reading this right?

| id | name | original_blueprint_id | active_version |
| --- | --- | --- | --- |
| 1 | Blueprint v1 | nil | false |
| 2 | Blueprint v1.1 | 1 | false |
| 3 | Blueprint v2 | 1 | false |
| 4 | Blueprint v2.1 | 1 | true |

My suggestion:

| id | name | ancestry | active_version (if we can assume the active_version is always the latest, then we don't need a column) |
| --- | --- | --- | --- |
| 1 | Blueprint v1 | nil | false |
| 2 | Blueprint v1.1 | 1 | false |
| 3 | Blueprint v2 | 1/2 | false |
| 4 | Blueprint v2.1 | 1/2/3 | true |